runtime/compiler/env/VMJ9.cpp

```diff
@@ -6988,6 +6988,17 @@ TR_J9VMBase::jitFieldsAreSame(TR_ResolvedMethod * method1, I_32 cpIndex1, TR_Res
    TR::VMAccessCriticalSection jitFieldsAreSame(this);
    bool result = false;
 
+   // Hidden classes generated within the same host class do not have distinct class names,
+   // but share the same field names with different field data types and offsets. Therefore,
+   // name-based check for whether fields are same can result in false positives when it comes
+   // to hidden classes unless the fields are from the same j9class objects.
+   if (method1->classOfMethod()
+       && method2->classOfMethod()
+       && (isHiddenClass(method1->classOfMethod())
+          || isHiddenClass(method2->classOfMethod()))
+       && method1->classOfMethod() != method2->classOfMethod())
+      return false;
```

Comment:
I'm not at all familiar with the use of `jitFieldsAreSame`, and I'm still reviewing, but my first question is whether other related methods need to include a similar check for hidden classes.  For instance, in [VMJ9Server::jitFieldsAreSame](https://github.com/eclipse-openj9/openj9/blob/24c9d7af6b89aee8ff1dcabd57e617db7602dce4/runtime/compiler/env/VMJ9Server.cpp#L466-L484).

Reply:
I have seen `jitFieldsAreSame` used in 2 places - ILGen (during shadow symbol lookups), and EscapeAnalysis.

As for `TR_J9ServerVM::jitFieldsAreSame`, it seems like this would also return false positives for hidden classes. I will implement the changes for handling hidden classes in JIT server mode as well.

Reply:
Unfortunately moving the check to `TR_ResolvedJ9/J9JITServerMethod::fieldsAreSame` would also require setting `sigSame` to `false` to guarantee that there will be no false positives, which is a lie because the field/class signatures do match for distinct hidden classes.